src/test_util/mod.rs

```diff
@@ -309,22 +309,32 @@ where
         .unwrap()
 }
 
-pub async fn wait_for<F, Fut>(mut f: F)
+pub async fn wait_for<F, Fut>(mut f: F, wait_for_secs: u64)
 where
     F: FnMut() -> Fut,
     Fut: Future<Output = bool> + Send + 'static,
 {
     let started = Instant::now();
     while !f().await {
         delay_for(Duration::from_millis(5)).await;
-        if started.elapsed().as_secs() > 5 {
+        if started.elapsed().as_secs() > wait_for_secs {
             panic!("Timed out while waiting");
         }
     }
 }
 
+// Wait for TCP socket to accept traffic-- defaults to 5 seconds
 pub async fn wait_for_tcp(addr: SocketAddr) {
```

Comment:
To avoid changing the signature of any functions already calling `wait_for_tcp`, I defaulted this to 5 seconds to mimic the current behavior.